### ISSUE TITLE
Texts including axis labels, tooltips, toolbox and legend are not rendered.

### ISSUE THREAD

**Issue by nipunadodan** [wuba/react-native-echarts#174]
**Describe the bug**
Texts including axis labels, tooltips, toolbox and legend are not rendered.

**To Reproduce**
Steps to reproduce the behaviour:
- Try a minimal example using SkiaChart with the below library versions.
```
"@wuba/react-native-echarts": "^1.3.1",
"@shopify/react-native-skia": "^0.1.241",
"echarts": "^5.5.0",
"react": "18.2.0",
"react-native": "^0.73.4",
```

**Expected behaviour**
Texts on axis labels, tooltips, legends and toolbox should be visible.

**Screenshots**
![image](https://github.com/wuba/react-native-echarts/assets/20305346/d321e1f9-cde3-44ad-9049-e4030cf79ac5)

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser: N/A
 - Version: 22.04.4

**Smartphone (please complete the following information):**
 - Device: Samsung A12
 - OS: Android 13
 - Browser: N/A
 - Version: N/A

**Additional context**
N/A

**Comment by DiyarYelbaka** [comment]
I have the same problem too. I hope it gets resolved as soon as possible.

**Comment by zhiqingchen** [comment]
use rnsvg or lower version of rnskia

**Comment by zhiqingchen** [comment]
https://github.com/Shopify/react-native-skia/issues/2404

**Comment by AlejandroRM-DEV** [comment]
> use rnsvg or lower version of rnskia

I switched to SvgChart and it worked, but is there any implication I should be aware of?

Thank you.

**Comment by zhiqingchen** [comment]
https://github.com/wuba/react-native-echarts/issues/16
@AlejandroRM-DEV

**Comment by nipunadodan** [comment]
> > use rnsvg or lower version of rnskia
> 
> I switched to SvgChart and it worked, but is there any implication I should be aware of?
> 
> Thank you.

Yes, I'm aware. SvgCharts works but it's slow to respond to gestures. That's why I implicitly need SkiaChart to work.

**Comment by Jay2009** [comment]
@nipunadodan  yes I am having exactly same issue with you. 
so any solution yet?

**Comment by liamjones** [comment]
@Jay2009 Below is what we've done as a workaround for now. It's not ideal but might help you (as long as you don't need react-native-skia for anything bar react-native-echarts on iOS).

@shopify/react-native-skia 0.1.234 is the last version that will render text in react-native-echarts. That version will crash/lock up an iOS app on RN 0.73.8 (the version of RN we're using, I haven't tried 0.74.x yet) but seems to work fine on Android. We want Skia rendering for slow Android devices as it's faster than SVG rendering. The iOS devices will use SVG rendering for now (since they have generally stronger performance).

How to set this up:

* Downgrade @shopify/react-native-skia to 0.1.234 in your `package.json`.
* Disable auto-linking of the module on iOS only by putting the following in your `react-native.config.js` (there's no point linking it if we can't use it):
```js
module.exports = {
    dependencies: {
        '@shopify/react-native-skia': {
            platforms: {
                ios: null
            }
        }
    }
}
```
* Update our React component to use either `SkiaChart` or `SvgChart` depending on platform:
```js
const SomeComponent = () => {
    const PlatformSpecificChartTag = Platform.OS === 'ios' ? SvgChart : SkiaChart
    return <PlatformSpecificChartTag option={...} etc />
}

```

**Comment by liamjones** [comment]
Now that [react-native-skia have documented that text in ImageSVG tags is not going to be supported](https://shopify.github.io/react-native-skia/docs/images-svg#text) could we add that to the known issues on https://github.com/wuba/react-native-echarts/issues/16?

**Comment by hthieu1110** [comment]
I got the same issue, all Legend + Axises are blank, do we have a plan to fix that or is there a patch/hack in the meaning time ?

**Comment by zhiqingchen** [comment]
We can't use it until we do a big [refactoring](https://github.com/wuba/react-native-echarts/issues/190), or

1. You can also use the old version of skia.
2. Use react-native-svg for rendering

@hthieu1110

**Comment by hthieu1110** [comment]
thankss @zhiqingchen , in our case we have to enable RN New Architect which is not supported by old version of Skia.
So react-native-svg is a working option (but is quite slow in comparison with Skia). So we are looking forward for the new-refactored version :D

Thanks @zhiqingchen again for your reply.

**Comment by zhiqingchen** [comment]
I am working on the [2.x](https://github.com/wuba/react-native-echarts/tree/2.x) branch, and there has been some progress, but there is still a lot of work to be done.

If you are interested, please review the code and provide me with some feedback.

<img src="https://github.com/user-attachments/assets/ffb62e2b-0f35-4f62-a1d7-6f803a5bdced" width="300px" />

Rendered by skia above and rendered by svg below, skia also supports shadow, which is a nice feature.

**Comment by ducpt-bili** [comment]
hi @zhiqingchen , how can i try the 2.x branch, i can test it for you.

**Comment by zhiqingchen** [comment]
1. Clone this repo
2. switch to 2.x branch 
3. yarn
4. cd example 
5. yarn android/ios
6. debug in src/example dir

@ducpt-bili

**Comment by ducpt-bili** [comment]
Hi @zhiqingchen ,
I did like you said and here is the error.
1. OK
2. OK
3. yarn => fail with multiple error of TS.
4. OK
5. Error like image below
<img width="1070" alt="Screenshot 2024-08-07 at 22 12 32" src="https://github.com/user-attachments/assets/790a5a94-3430-4a25-9ba4-2f7a6ce95192">

**Comment by ducpt-bili** [comment]
hi @zhiqingchen , any news on the version 2. Thanks.

**Comment by zhiqingchen** [comment]
dev version
2.0.0-canary.0

sample code
https://github.com/wuba/react-native-echarts/blob/2.x/example/src/App.tsx

**Comment by ducpt-bili** [comment]
hi @anon-r-7 , did you follow the @zhiqingchen  guide and have the same problem like me?

**Comment by ducpt-bili** [comment]
hi @zhiqingchen , i able to run your 2.x example successfully, but it have some bug:
1. On android, text is showing, but tool tip not working on Skia chart.
2. On ios, text is NOT shoing, but tooltop is working on Skia chart.
Thank you for your hard work.

**Comment by zhiqingchen** [comment]
@ducpt-bili thanks. Please provide the code of the bug.

**Comment by ducpt-bili** [comment]
hi @zhiqingchen , i just run the code in the example folder on 2.x branch, not add any thing new to it. 
Both android and ios i run is in simulator or emulator.

**Comment by ducpt-bili** [comment]
hi @zhiqingchen , is there anything i can help for testing?

**Comment by zhiqingchen** [comment]
https://github.com/wuba/react-native-echarts/pull/193
Related discussion moved to this PR, your previous feedback question we have not repeated.https://github.com/user-attachments/assets/c3ab96f8-3320-4406-9699-cdd9bb02005a

**Comment by ducpt-bili** [comment]
hi @zhiqingchen , thank you. 
Maybe i will wait for version 2 release to make sure my setup is up-to-date. Right now maybe some mismatch setup will happen.

Thank you again.

**Comment by zhiqingchen** [comment]
try 2.0.0-canary.3

**Comment by ducpt-bili** [comment]
@zhiqingchen let me take a quick test. Thank you.

**Comment by ducpt-bili** [comment]
I met this error below on ios simulator 17.4 when try 2.0.0-canary.3, with Skia chart.
Note: it work with previous version on Skia and when i change to SVG chart it work.

<img width="350" alt="Screenshot 2024-10-11 at 16 52 15" src="https://github.com/user-attachments/assets/4e10fd27-94be-43fb-90d6-86b3b5170d5b">

**Comment by zhiqingchen** [comment]
try this demo 
https://github.com/zhiqingchen/react-native-echarts-test
![image](https://github.com/user-attachments/assets/59a6f092-7ef2-4942-a570-ac34628e6eb3)

**Comment by ducpt-bili** [comment]
hi @zhiqingchen , i try out your repo and it work. Thank you.
The problem is from the renderer:
renderer: "svg" => Error like mine above.
renderer: "skia" => WORK
Looking forward for the stable release :D Your library have potential to be the number one library of charts in React Native. Only one chart to be competitive is victory native xl, but it have some limit like now horizontal scroll or limit type of chart.
Thank you for your hard work.

**Comment by zhiqingchen** [comment]
https://wuba.github.io/react-native-echarts/docs/migration/v2

**Comment by zhiqingchen** [comment]
yes, and [expo snacks](https://wuba.github.io/react-native-echarts/docs/expo-snacks/simple-line-chart) are ready too.

**Comment by AleeFarook** [comment]
@zhiqingchen  I want to show image in tooltip, how I am doing this using richtext but it is not working in SvgChart, Please help 

```
tooltip: {
          trigger: 'item',
          confine: true,
          axisPointer: {
            type: 'cross',
            label: {
              color: '#000000',
              fontSize: 11,
              backgroundColor: 'rgba(230, 230, 230, 1)',
              formatter: (params) => {
                if (params.axisDimension === 'x') {
                  return addCommas(Math.round(params.value)) + " KM";
                } else if (params.axisDimension === 'y') {
                  return addCommas(Math.round(params.value)) + " AED";
                }
              },
            },
          },
          rich: {
            image: {
              width: 50,
              height: 50,
              align: 'center',
              backgroundColor: {
                image: 'https://letsenhance.io/static/73136da51c245e80edc6ccfe44888a99/1015f/MainBefore.jpg', 
              },
            },
          },
          formatter: (params) => tooltipFormatter(params),
        },
        
            function tooltipFormatter(params) {
      if (params.seriesType === 'scatter') {
        if (params.seriesName === "Rec. trade-in price") {
          const [mileage, price] = params.data;
          return `{centerText|Recommended Price ${price}}`;
        } else {
          const [mileage, price, make, model, adImage, year] = params.data;
          return `{image|}\n{info|${year} ${make} ${model}}\n{price|${addCommas(price)} AED}\n{mileage|${addCommas(mileage)} KM}`;  
        }
      } else if (params.seriesType === 'line') {
        return `{centerText|${params.seriesName}}`;
      }
      return '';
    }


```
